chainstate/src/detail/chainstateref.rs

```diff
@@ -136,25 +137,54 @@ impl<'a, S: BlockchainStorageRead, O: OrphanBlocks> ChainstateRef<'a, S, O> {
             .ok_or_else(|| PropertyQueryError::PrevBlockIndexNotFound(prev_block_id.clone()))
     }
 
-    // TODO improve using pskip
     pub fn get_ancestor(
         &self,
         block_index: &BlockIndex,
-        ancestor_height: BlockHeight,
+        target_height: BlockHeight,
     ) -> Result<BlockIndex, PropertyQueryError> {
-        if ancestor_height > block_index.block_height() {
+        if target_height > block_index.block_height() {
             return Err(PropertyQueryError::InvalidAncestorHeight {
                 block_height: block_index.block_height(),
-                ancestor_height,
+                ancestor_height: target_height,
             });
         }
 
+        let step_to_prev_block = |block_index_walk: &mut BlockIndex,
+                                  height_walk: &mut BlockHeight|
+         -> Result<(), PropertyQueryError> {
+            *block_index_walk = self.get_previous_block_index(block_index_walk)?;
+            *height_walk = (*height_walk - BlockDistance::from(1))
+                .expect("height_walk is greater than height");
```

Comment:
I added `BlockHeight::prev_height` but it's a part of a live PR #246, not merged yet. Slightly more convenient and works in more cases.